src/main.c

```diff
@@ -128,9 +128,17 @@ main (int argc, char *argv[])
   }
   while (next_option != -1);
 
-  bufchk (getenv ("HOME"), MP);
+  if (getenv ("HOME") == NULL)
+  {
+    fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
+    return 1;
+  }
+  
   char HOMEDIR[strlen (getenv ("HOME")) + 1];
```

Comment:
but when getenv is called it already reserved some space in memory ..

check if you like the change or else it is ok : 
```c
 char *HOMEDIR = getenv ("HOME");
 if (HOMEDIR == NULL)
  {
    fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
    return 1;
  }
 
  if (bufchk (HOMEDIR, MP))
  return BUF_ERR;
```